guide/expressions.md

```diff
@@ -253,6 +253,35 @@ Do not put any spaces around the `.`.
 x.foo().bar().baz(x, y, z);
 ```
 
+
+### Macro uses
+
+Macros which can be parsed like other constructs should be formatted like those
+constructs. For example, a macro use `foo!(a, b, c)` can be parsed like a
+function call (ignoring the `!`), therefore it should be formatted following the
+rules for function calls.
+
+#### Special case macros
+
+Macros which take a format string and where all other arguments are *small* may
```

Comment:
Does that mean that `write!`  must be formatted like this:

```
write!(
    f,
    "Hello World, {}!",
    name
);
```

I'd prefer to allow a single parameter before the format string on the same line.

Reply:
That's right. It can still be formatted on one line if it is short enough, the alternative is that every paramater gets its own line, so this is a more condensed version in practice. When we do go multi-line, it is usually because the string literal is long, so putting `f` on its own line it probably a good move.